src/librustc_mir/mir_map.rs

```diff
@@ -148,6 +148,10 @@ impl<'a, 'm, 'tcx> Visitor<'tcx> for InnerDump<'a,'m,'tcx> {
 
         match build_mir(Cx::new(&infcx), implicit_arg_tys, id, span, decl, body) {
             Ok(mut mir) => {
+                // FIXME: This should run later rather than earlier (since this is supposed to be a
```

Comment:
This doesn't seem like a FIXME necessarily. Once we are done with type-checking and other things, the MIR is basically ripe for optimization in any order, and taking into account codegen options seems legit.